### PR TITLE
[PM-12922] Disable delete if user can't manage collection 

### PR DESCRIPTION
**Pull request by SaintPatrck**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-12992

## 📔 Objective

Disables the delete button for items in collections where the user does not have "manage" permission.

This change ensures that users cannot delete items from collections they are not authorized to manage. It updates the UI to reflect the user's permissions and prevents accidental or unauthorized deletions.

Depends on #4169 

## 📸 Screenshots

| Before | After |
|--------|--------|
| image | image | 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditScreen.kt [hunk 1]
```diff
@@ -235,7 +235,7 @@ fun VaultAddEditScreen(
         },
     )
 
-    if (pendingDeleteCipher) {
+    if (pendingDeleteCipher && state.canDelete) {
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditScreen.kt:238]: We should not need this, right?

The other check should handle it all

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditViewModel.kt [hunk 2]
```diff
@@ -2129,6 +2151,7 @@ data class VaultAddEditState(
                 val availableFolders: List<Folder> = emptyList(),
                 val selectedOwnerId: String? = null,
                 val availableOwners: List<Owner> = emptyList(),
+                val canDelete: Boolean = true,
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditViewModel.kt:2154]: Wanna add a doc for this

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/item/VaultItemScreen.kt [hunk 3]
```diff
@@ -163,7 +163,7 @@ fun VaultItemScreen(
                     { viewModel.trySendAction(VaultItemAction.Common.CloseClick) }
                 },
                 actions = {
-                    if (state.isCipherDeleted) {
+                    if (state.isCipherDeleted && state.canDelete) {
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/item/VaultItemScreen.kt:166]: So you can only restore a cipher if it can be deleted?

Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/item/VaultItemScreen.kt:166]: Maybe it should be `canManage` to be more generic?

Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/item/VaultItemScreen.kt:166]: Ah. Nope. This check shouldn't be here.

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditScreen.kt (hunk 1 above)
Comment (SaintPatrck) [reply app/src/main/java/com/x8bit/bitwarden/ui/vault/feature/addedit/VaultAddEditScreen.kt:238]: You're right!

**Review by david-livefront — DISMISSED** [review]

**Review by david-livefront — DISMISSED** [review]

**Review by david-livefront — DISMISSED** [review]

File: app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/util/TotpDataExtensionsTest.kt [hunk 4]
```diff
@@ -42,7 +42,9 @@ class TotpDataExtensionsTest {
                 isIndividualVaultDisabled = false,
                 type = VaultAddEditState.ViewState.Content.ItemType.Login(totp = uri),
             ),
-            totpData.toDefaultAddTypeContent(isIndividualVaultDisabled = false),
+            totpData.toDefaultAddTypeContent(
+                isIndividualVaultDisabled = false,
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/util/TotpDataExtensionsTest.kt:46]: Did you intend to change this file?

Comment (SaintPatrck) [reply app/src/test/java/com/x8bit/bitwarden/ui/vault/feature/addedit/util/TotpDataExtensionsTest.kt:46]: Nope.

**Review by david-livefront — DISMISSED** [review]

**Review by david-livefront — APPROVED** [review]

**Pull request merged**